### PR TITLE
Update prereqs per current portability testing

### PR DESCRIPTION
**Pull request by mppf**
This PR updates prereqs.rst according to the current output of `extract-docs.py`.

(Note that PR #21829 includes the portability testing changes that were used to generate this documentation change).

Reviewed by @tzinsky and @lydia-duncan - thanks!

### CODE REVIEW

**Comment by tzinsky** [comment]
Thanks for the updates to these dependency lists.   My responses while you wait for others... 
> 
> Questions for reviewer(s):
> 
> * should we keep the Amazon Linux 2 lines? They are a bit long

I know they are long but they do support system LLVM and would be an option for folks while Amazon Linux 2023 gets supported. 

> * should we keep the Amazon Linux 2023 and Fedora 37/38 lines here? These configurations don't have a working CHPL_LLVM=system, yet

I would omit these variants until they support a system LLVM.   

> * should we keep the Homebrew line? Here it is tested only with linux brew but the commands look right

I say we keep it unless we have something specific to homebrew elsewhere.

**Review by tzinsky — APPROVED** [review]

**Comment by mppf** [comment]
> > * should we keep the Homebrew line? Here it is tested only with linux brew but the commands look right
> 
> I say we keep it unless we have something specific to homebrew elsewhere.

We do, but it's for Mac OS X, and we tell people to just install the Chapel Homebrew package:

https://chapel-lang.org/docs/platforms/macosx.html

Which is different from getting the prerequisites installed.

File: doc/rst/usingchapel/prereqs.rst [hunk 1]
```diff
@@ -152,19 +167,31 @@ We have used the following commands to install the above prerequisites:
...
+
+  * FreeBSD 12.2, 12.4, 13.1::
 
       sudo pkg install gcc m4 perl5 python3 bash gmake gawk git pkgconf cmake
       sudo pkg install llvm13
 
 
+  * Homebrew::
+
+      brew install cmake python gmp llvm@14
```
Comment (lydia-duncan) [inline doc/rst/usingchapel/prereqs.rst:185]: The homebrew line seems a bit weird considering everything else on this list is an operating system, right?  Maybe it should list Mac OS versions instead?

Comment (mppf) [reply doc/rst/usingchapel/prereqs.rst:185]: Right, but in terms of "how do I install dependencies", it is answering "How do I install the dependencies on Homebrew". Note that Homebrew can run on linux, so it's not necessarily the case that somebody interested in Homebrew is using a Mac (or that they will look in a Mac section).

If we listed Mac OS X versions, I would also expect the instructions to describe how to install Homebrew, which is not something I wanted to get in to. But, anyway, we already have documentation about Mac OS X that suggests `brew install chapel` which makes these prerequisites irrelevant, in that use case.

I'm half a mind to just remove this section so if it is a source of contention I'll just remove it. We didn't seem to need it in the last release.

**Review by lydia-duncan — APPROVED** [review]
Thanks!

**Pull request merged**